api/extension-guides/language-model-tutorial.md

```diff
@@ -9,13 +9,324 @@ MetaDescription: Tutorial that walks you through creating a VS Code extension th
 
 # Tutorial: Generate AI-powered code annotations by using the Language Model API
 
-In this tutorial, you'll learn ...
+In this tutorial, you'll learn how to create a Visual Studio Code extension that uses the Language Model API to generate AI-powered code annotations. You'll create a Code Tutor extension that provides inline suggestions for how to improve your code.
+
+<!-- TODO: Add a screenshot of the extension in action -->
+![]()
 
 ## Prerequisites
 
-## Step 1: do something
+You'll need the following tools and accounts to complete this tutorial:
+
+- [Visual Studio Code](https://code.visualstudio.com/download)
+- [Node.js](https://nodejs.org/en/download/)
+
+## Step 1: Scaffold out the extension
+
+First, use Yeoman and VS Code Extension Generator to scaffold a TypeScript or JavaScript project ready for development.
+
+```bash
+npx --package yo --package generator-code -- yo code
+```
+
+Select the following options to complete the new extension wizard...
+
+```bash
+# ? What type of extension do you want to create? New Extension (TypeScript)
+# ? What's the name of your extension? Code Tutor
+
+### Press <Enter> to choose default for all options below ###
+
+# ? What's the identifier of your extension? code-tutor
+# ? What's the description of your extension? LEAVE BLANK
+# ? Initialize a git repository? Yes
+# ? Bundle the source code with webpack? No
+# ? Which package manager to use? npm
+
+# ? Do you want to open the new folder with Visual Studio Code? Open with `code`
+```
+
+## Step 2: Modify the package.json file to include the correct commands
+
+By default, the scaffolded project will include a single "helloWorld" command in the `package.json` file. This command will show up in the Command Palette when your extension is installed. You'll want to modify this so that it shows a "Show Code Tutor Annotations" command instead.
+
+In the `package.json` file, find the "contributes" section and modify the existing "helloWorld" command so that it looks like this:
+
+```json
+"contributes": {
+  "commands": [
+      {
+      "command": "code-tutor.annotate",
+      "title": "Tutor"
+      }
+  ]
+}
+```
+
+We also want a "Tutor" button to show up at the top of the editor so that the user can easily activate the tutor annotations. To do that, add a "menus" section in the "contributes" section, just under the "commands".
+
+```json
+"contributes": {
+  "commands": [
+    {
+      "command": "code-tutor.annotate",
+      "title": "Tutor"
+    }
+  ],
+  "menus": {
+    "editor/title": [
+      {
+        "when": "editorFocus",
+        "command": "code-tutor.annotate",
+        "group": "navigation"
+      }
+    ]
+  }
+}
+```
+
+- The "editor/title" specifies where we want this menu option - which is in the title bar of whichever tab is currently open in the editor area.
+- The "when" specifies when we want this menu option to be shown. In this case, we want it to be shown whenever the editor has focus.
```

Comment:
Consider not using the when clause at all to simplify.
Also doing when for focus will make the button disappearing thus not aligning with the other buttons in the editor title area.